web/html/src/utils/data-providers/page-control.ts

```diff
@@ -0,0 +1,29 @@
+type SortType = {
+    direction: number;
+    column: string;
+};
+
+export default class PageControl {
+    page: number;
+    pageSize: number;
+    query: string | null | undefined;
```

Comment:
I think this is usually not the use case we actually want to go for, I simply didn't change anything here as I want to not touch things functionality-wise for now and just migrate. Later on we can fix any issues with confidence in the refactoring with strict types.  

For example, in the above case, I would say that `query` should probably just be `query?: string`, since I doubt a `null` query is functionally different from an `undefined` or empty string one.  

Once we've migrated the whole code base, I'd like to do a followup for these types of issues and other similar ones, enabling stricter typing across the board. But for the first version, I would simply go with what we can currently infer.

Reply:
Sure, at this point I also cannot say either `null` or `undefined` for any of these.

What I mean is something that has the same effect as `string | null | undefined` but looks better like:
```
type Optional<T> = T | null | undefined;
```

Reply:
Such a type could be defined and made available of course, but I think it's an anti-pattern. There are cases where `T | null | undefined` makes sense (such as dropdowns where `null` can denote a chosen empty value and `undefined` no chosen value etc), but most of the time it would be better to look for more constrained types.  

I would argue against creating such a type because adding it makes its use easy. Instead, I think specifying `T | null | undefined` should intentionally be annoying so that the developer considers whether they can maybe use fewer types instead. Hope my reasoning makes sense, let me know if I wasn't very clear.  

In short, I would advise against such a generic to intentionally reduce the use of such types.

Reply:
> I would argue against creating such a type because adding it makes its use easy. Instead, I think specifying T | null | undefined should intentionally be annoying so that the developer considers whether they can maybe use fewer types instead.

I was thinking of this just as a temporary thing to target such occurrences that are already there, but this statement, I also agree. Leaving it like this would encourage us to fix these as we go. :+1: